tests/lean/run/timeFormats.lean

```diff
@@ -0,0 +1,785 @@
+
+/--
+info: "02:11:01 PM"
+-/
+#guard_msgs in
+#eval l
+/--
+info: "03:11:01 AM"
+-/
+#guard_msgs in
+#eval Time12Hour.formatBuilder time₂.hour.toRelative time₂.minute time₂.second (if time₂.hour.val > 12 then HourMarker.pm else HourMarker.am)
+
+/--
+info: "06/16/2014"
+-/
+#guard_msgs in
+#eval ShortDate.formatBuilder date₁.month date₁.day date₁.year
+
+/--
+info: "0053-06-19"
+-/
+#guard_msgs in
+#eval Formats.sqlDate.format (DateTime.ofPlainDate (PlainDate.ofDaysSinceUNIXEpoch ⟨-700000⟩) .UTC)
+
+/--
+info: "0003-09-16"
+-/
+#guard_msgs in
+#eval Formats.sqlDate.format (DateTime.ofPlainDate (PlainDate.ofDaysSinceUNIXEpoch ⟨-720000⟩) .UTC)
+
+/--
+info: "0085-07-28"
+-/
+#guard_msgs in
+#eval Formats.sqlDate.format (DateTime.ofPlainDate (PlainDate.ofDaysSinceUNIXEpoch ⟨-750000⟩) .UTC)
+
+/--
+info: "0222-09-04"
+-/
+#guard_msgs in
+#eval Formats.sqlDate.format (DateTime.ofPlainDate (PlainDate.ofDaysSinceUNIXEpoch ⟨-800000⟩) .UTC)
+
+/--
+info: date("0222-09-04")
+-/
+#guard_msgs in
+#eval PlainDate.ofDaysSinceUNIXEpoch ⟨-800000⟩
+
+/--
+info: date("0222-09-04")
+-/
+#guard_msgs in
+#eval PlainDate.ofDaysSinceUNIXEpoch ⟨-800000⟩
+
+/--
+info: date("2002-07-14")
+-/
+#guard_msgs in
+#eval date("2002-07-14")
+
+/--
+info: time("14:13:12.000000000")
+-/
+#guard_msgs in
+#eval time("14:13:12")
+
+/--
+info: zoned("2002-07-14T14:13:12.000000000Z")
+-/
+#guard_msgs in
+#eval zoned("2002-07-14T14:13:12Z")
+
+/--
+info: zoned("2002-07-14T14:13:12.000000000+09:00")
+-/
+#guard_msgs in
+#eval zoned("2002-07-14T14:13:12+09:00")
+
+/--
+info: "2002-07-14"
+-/
+#guard_msgs in
+#eval zoned("2002-07-14T14:13:12+09:00").format "yyyy-MM-dd"
+
+/--
+info: "14-13-12"
+-/
+#guard_msgs in
+#eval zoned("2002-07-14T14:13:12+09:00").format "HH-mm-ss"
+
+/-
+Format
+-/
+
+def time₄ := time("23:13:12.324354679")
+def date₄ := date("2002-07-14")
+def datetime₅ := PlainDateTime.mk (PlainDate.clip (-2000) 3 4) (PlainTime.mk 12 23 ⟨false, 12⟩ 0)
+def datetime₄ := datetime("2002-07-14T23:13:12.324354679")
+def zoned₄ := zoned("2002-07-14T23:13:12.324354679+09:00")
+
+/--
+info: "CE CE CE Common Era C"
+-/
+#guard_msgs in
+#eval zoned₄.format "G GG GGG GGGG GGGGG"
+
+/--
+info: "02 2002 000002002"
+-/
+#guard_msgs in
+#eval zoned₄.format "yy yyyy yyyyyyyyy"
+
+/--
+info: "02 2002 000002002"
+-/
+#guard_msgs in
+#eval zoned₄.format "uu uuuu uuuuuuuuu"
+
+/--
+info: "5 95 195"
+-/
+#guard_msgs in
+#eval zoned₄.format "D DD DDD"
+
+/--
+info: "7 07 Jul J"
+-/
+#guard_msgs in
+#eval zoned₄.format "M MM MMM MMMMM"
+
+/--
+info: "4 14 014 0014 00014"
+-/
+#guard_msgs in
+#eval zoned₄.format "d dd ddd dddd ddddd"
+
+/--
+info: "7 07 Jul July J"
+-/
+#guard_msgs in
+#eval zoned₄.format "M MM MMM MMMM MMMMM"
+
+/--
+info: "4 14 0014 0014"
+-/#guard_msgs in
+#eval zoned₄.format "d dd dddd dddd"
+
+/--
+info: "3 03 3rd quarter 3"
+-/
+#guard_msgs in
+#eval zoned₄.format "Q QQ QQQQ QQQQQ"
+
+/--
+info: "8 28 028 0028"
+-/
+#guard_msgs in
+#eval zoned₄.format "w ww www wwww"
```

Comment:
Java informs me that the correct answer here is in fact `29`.

Reply:
Java starts a week on Sunday and the library starts it on Monday. I'm going to add documentation for it.